blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -0,0 +1,192 @@
+
+  def withCheckEndpointAuthentication(checkEndpointIdentification: Boolean): BlazeClientBuilder[F] =
+    copy(checkEndpointIdentification = checkEndpointIdentification)
+
+  def withMaxResponseLineSize(maxResponseLineSize: Int): BlazeClientBuilder[F] =
+    copy(maxResponseLineSize = maxResponseLineSize)
+
+  def withMaxChunkSize(maxChunkSize: Int): BlazeClientBuilder[F] =
+    copy(maxChunkSize = maxChunkSize)
+
+  def withParserMode(parserMode: ParserMode): BlazeClientBuilder[F] =
+    copy(parserMode = parserMode)
+
+  def withBufferSize(bufferSize: Int): BlazeClientBuilder[F] =
+    copy(bufferSize = bufferSize)
+
+  def withExecutionContext(executionContext: ExecutionContext): BlazeClientBuilder[F] =
+    copy(executionContext = executionContext)
+
+  def withAsynchronousChannelGroupOption(
+      asynchronousChannelGroup: Option[AsynchronousChannelGroup]): BlazeClientBuilder[F] =
+    copy(asynchronousChannelGroup = asynchronousChannelGroup)
+  def withAsynchronousChannelGroup(
+      asynchronousChannelGroup: AsynchronousChannelGroup): BlazeClientBuilder[F] =
+    withAsynchronousChannelGroupOption(Some(asynchronousChannelGroup))
+  def withoutAsynchronousChannelGroup: BlazeClientBuilder[F] =
+    withAsynchronousChannelGroupOption(None)
+
+  def resource(implicit F: ConcurrentEffect[F]): Resource[F, Client[F]] =
+    Resource.make(
+      connectionManager.map(
+        manager =>
+          BlazeClient.makeClient(
+            manager = manager,
+            responseHeaderTimeout = responseHeaderTimeout,
+            idleTimeout = idleTimeout,
+            requestTimeout = requestTimeout
+        )))(_.shutdown)
+
+  def stream(implicit F: ConcurrentEffect[F]): Stream[F, Client[F]] =
+    Stream.resource(resource)
+
+  private def connectionManager(
+      implicit F: ConcurrentEffect[F]): F[ConnectionManager[F, BlazeConnection[F]]] = {
+    val http1: ConnectionBuilder[F, BlazeConnection[F]] = new Http1Support(
+      sslContextOption = sslContext,
+      bufferSize = bufferSize,
+      asynchronousChannelGroup = asynchronousChannelGroup,
+      executionContext = executionContext,
+      checkEndpointIdentification = checkEndpointIdentification,
+      maxResponseLineSize = maxResponseLineSize,
+      maxHeaderLength = maxHeaderLength,
+      maxChunkSize = maxChunkSize,
+      parserMode = parserMode,
+      userAgent = userAgent
+    ).makeClient
+    ConnectionManager
+      .pool(
+        builder = http1,
+        maxTotal = maxTotalConnections,
+        maxWaitQueueLimit = maxWaitQueueLimit,
+        maxConnectionsPerRequestKey = maxConnectionsPerRequestKey,
+        responseHeaderTimeout = responseHeaderTimeout,
+        requestTimeout = requestTimeout,
+        executionContext = executionContext
+      )
+  }
+}
+
+object BlazeClientBuilder {
+  def apply[F[_]](
+      executionContext: ExecutionContext,
+      sslContext: Option[SSLContext] = Some(SSLContext.getDefault)): BlazeClientBuilder[F] =
```

Comment:
We have to treat this one delicately because it can throw in grim circumstances.  See #1957.

Reply:
In what situation does it throw that it shouldn't be a fatal uncaught error?